pkg/datapath/loader/loader.go

```diff
@@ -140,15 +176,57 @@ func nullifyStringSubstitutions(strings map[string]string) map[string]string {
 	return nullStrings
 }
 
+func (l *loader) bpfMasqAddrs(ifName string) (masq4, masq6 netip.Addr) {
+	if l.cfg.DeriveMasqIPAddrFromDevice != "" {
+		ifName = l.cfg.DeriveMasqIPAddrFromDevice
+	}
+
+	find := func(iter statedb.Iterator[tables.NodeAddress]) bool {
+		for addr, _, ok := iter.Next(); ok; addr, _, ok = iter.Next() {
+			if ifName != defaults.HostDevice && addr.DeviceName == defaults.HostDevice {
```

Comment:
Why not the following, instead?

```
if ifName != addr.DeviceName {
  continue
}
```

Do we suspect that if not HostDevice, these strings are unlikely to match?

Reply:
Hmm I have no idea anymore why I added this there. The whole check isn't useful at all. Removed.